components/sup/src/http_gateway.rs

```diff
@@ -198,6 +265,25 @@ impl Server {
     }
 }
 
+fn tls_config(tls_files: Option<(path::PathBuf, path::PathBuf)>) -> Option<ServerConfig> {
+    match tls_files {
+        Some((key_file, cert_file)) => {
+            let mut config = ServerConfig::new(NoClientAuth::new());
+            let key_file = &mut BufReader::new(File::open(key_file).unwrap());
+            let cert_file = &mut BufReader::new(File::open(cert_file).unwrap());
```

Comment:
It's not safe to `unwrap` here because all that `file_exists` validates is presence, not readability. I'd suggest replacing the `file_exists` validator with a `file_readable` one and passing this function `std::fs::File` structs rather than `PathBuf`s.

Reply:
You're right about `file_exists` but I verify readability [here](https://github.com/habitat-sh/habitat/pull/5874/files#diff-cebe13127ff4a329c357419d517b1b65R256) where the CLI args are parsed into a `ManagerConfig` struct.

Reply:
Oh! I totally missed that. How about just populating `cfg.tls_files` in `mgrcfg_from_matches` and moving all the validation into `tls_config`? Just moving `validate_key_file` and `validate_cert_file` to `manager/mod.rs` makes this change very simple. I think it makes a bit more sense to do all the validation in one place and it seem like `mgrcfg_from_matches` is more about just populating that struct than digging into the values.

With that change, you can also make
```rust
    if kf.is_some() && cf.is_some() {
        let kpb = Path::new(kf.unwrap());
        let cpb = Path::new(cf.unwrap());

        validate_key_file(kpb.as_path())?;
        validate_cert_file(cpb.as_path())?;

        cfg.tls_files = Some((kpb, cpb));
    }
```
a bit more concise/rustic:
```rust
    if let (Some(kf), Some(cf)) = (m.value_of("KEY_FILE"), m.value_of("CERT_FILE")) {
        cfg.tls_files = Some((PathBuf::from(kf), PathBuf::from(cf)));
    }
```

Reply:
Will do